File: lib/handler/request_decoder.rb
class Katamari::Handler::RequestDecoder < Katamari::Handler::Upstream
  import org.jboss.netty.channel.Channels
  
  def messageReceived(ctx, e)
    Channels.fireMessageReceived(ctx, Katamari::Env.new(e))
  end
end

File: src/main/java/io/katamari/handler/RequestDecoder.java
package io.katamari.handler;

import io.netty.handler.codec.http.HttpMessage;
import io.netty.handler.codec.http.HttpVersion;
import io.netty.handler.codec.http.HttpMethod;
import io.netty.handler.codec.http.HttpRequestDecoder;

import io.katamari.env.Request;

public class RequestDecoder extends HttpRequestDecoder {
  @Override
  protected HttpMessage createMessage(String[] initialLine) throws Exception {
    return new Request(
      HttpVersion.valueOf(initialLine[2]), HttpMethod.valueOf(initialLine[0]), initialLine[1]);
  }
}